x/oracle/keeper/historic_price.go

```diff
@@ -34,6 +34,11 @@ func (k Keeper) CalcAndSetHistoricMedian(
 	denom string,
 ) error {
 	historicPrices := k.historicPrices(ctx, denom, k.MaximumPriceStamps(ctx))
+	// this should never happen
+	if len(historicPrices) == 0 {
+		return nil
```

Comment:
Let's add a log here with the err saying this should never be the case + a test for it

Reply:
done

Reply:
Maybe a more descriptive msg here

Reply:
Could you provide some inside on why this shouldn't happen never? I'm not so related with the x/oracle module, that's why the message it's not so descriptive.

Reply:
this only happens when the historic stamp period is less than the median stamp period; this doesn't make any sense since the medians need to be calculated based off the existing historic stamps. 

It should never happen since there should always be historic stamps before we attempt to make any median stamps

Reply:
gotchu, improved the log